one_fm/operations/doctype/operations_post/operations_post.py

```diff
@@ -14,171 +14,185 @@
+def create_new_schedule_for_project(proj):
+    # Generate new post schedules for single project that have been generated by the roster projection view report
+    try:
+        existing_proj = frappe.db.exists("Project",proj)
+        if existing_proj:
+            all_operations_post = frappe.get_all("Operations Post",{'project':existing_proj})
+            all_operations_post_ = [frappe.get_doc("Operations Post",i.name) for i in all_operations_post]            
+            frappe.enqueue(create_post_schedules, operations_posts=all_operations_post_, queue="long",job_name = 'Create Post Schedules')
+            return response("Post Creation Scheduled Sucessfully",{}, True, 200)
+    except:
+        frappe.log_error("Post Schedule Creation Error",frappe.get_traceback())
+    
 
 
 def execute_post_creation():
-	"""
-		Get the list of projects with 0 Post Schedules  from the roster projection view report and Create the post schedules for projects with 0 
-	"""
-	filters = frappe._dict({
-		'month':'06',
-		'year':'2023'
-	})
-	result = report_execute(filters=filters)
-	if result and len(result) > 1:
-		project_set = set()
-		for each in result[1]:
-			if each.ps_qty in [0.0,0]:
-				project_set.add(each.get('project'))
-		update_ops_post(project_set)
-		
-		
-	
-	
+    """
+        Get the list of projects with 0 Post Schedules  from the roster projection view report and Create the post schedules for projects with 0 
+    """
+    filters = frappe._dict({
+        'month':'06',
+        'year':'2023'
+    })
+    result = report_execute(filters=filters)
+    if result and len(result) > 1:
+        project_set = set()
+        for each in result[1]:
+            if each.ps_qty in [0.0,0]:
+                project_set.add(each.get('project'))
+        update_ops_post(project_set)
+        
+        
+    
+    
 
 def update_ops_post(projects):
-	try:
-		tuple_projects = tuple(projects)
-		query = (f"Select name from `tabOperations Post` where project in {tuple_projects}")
-		response = frappe.db.sql(query,as_dict=1)
-		if response:
-			posts = [i.name for i in response]
-	except:
-		frappe.log_error("Post Schedule Creation Error",frappe.get_traceback())
-	
-	try:
-		for each in posts:
-			doc = frappe.get_doc("Operations Post",each)
-			if doc.project == 'Al-Khorayef':
-				pass
-			create_post_schedule_for_operations_post(doc)
-			frappe.db.commit()
-	except:
-		frappe.log_error("Error Creating Post Schedules",frappe.get_traceback())
+    try:
+        tuple_projects = tuple(projects)
+        query = (f"Select name from `tabOperations Post` where project in {tuple_projects}")
+        response = frappe.db.sql(query,as_dict=1)
+        if response:
+            posts = [i.name for i in response]
+    except:
+        frappe.log_error("Post Schedule Creation Error",frappe.get_traceback())
+    
+    try:
+        for each in posts:
+            doc = frappe.get_doc("Operations Post",each)
+            if doc.project == 'Al-Khorayef':
```

Comment:
Hard code a project name is not acceptable. Instead of the project name check the condition here to pass the execution.

Reply:
Thank you @pjamsheer  It was a line for debugging. I have removed it